portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,22 +32,33 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
   private Gson gson = new Gson();
-  private List<String> listOfComments = new ArrayList<String>();
+  private Query query = new Query("Comment");
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get comments from datastore
+    PreparedQuery results = datastore.prepare(query);
+    List<String> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      comments.add((String)entity.getProperty("content"));
+    }
+
     // Send json as the response
-    String json = convertToJsonUsingGson(listOfComments);
+    String json = convertToJsonUsingGson(comments);
     response.setContentType("application/json");
     response.getWriter().println(json);
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String comment = getParameter(request, "text-input", "");
-    listOfComments.add(comment);
-
+    String comment = request.getParameter("text-input");
+    // Add comment entity to Datastore
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("content", comment);
```

Comment:
Since the entity and property names ("Comment" and "content") are referenced in multiple places that need to stay in-sync, it's best to make them constants (ensures that typos are caught at compile time).

Reply:
Makes sense. I'll make the change!